Source/Session.swift

```diff
@@ -204,7 +204,29 @@ open class Session {
         session.invalidateAndCancel()
     }
 
-    // MARK: - Cancellation
+    // MARK: - All Requests API
+
+    /// Perform an action on all active `Request`s, optionally calling a completion handler when complete.
+    ///
+    /// - Note: The provided `action` closure is performed asynchronously, meaning that some `Request`s may complete and
+    ///         be unavailable by time it runs. Additionally, these actions are performed on the instances's `rootQueue`,
+    ///         so care should be taken that actions are fast.
+    ///
+    /// - Parameters:
+    ///   - queue:      `DispatchQueue` on which the completion handler is called. `.main` by default.
+    ///   - action:     Closure to perform on each `Request`.
+    ///   - completion: Closure to be called when `action` has been called on all `Request`s. `nil` by default.
+    public func withAllRequests(completingOn queue: DispatchQueue = .main,
+                                perform action: @escaping (Request) -> Void,
+                                completion: (() -> Void)? = nil) {
```

Comment:
This API doesn't really match our other APIs conventions with `completion` closures. Would the following make more sense?

```swift
public func withAllRequests(
    perform action: @escaping (Request) -> Void,
    queue: DispatchQueue = .main,
    completion: (() -> Void)? = nil)
```

This would also put the parameters with default values at the end of the signature. Did you pivot from the convention since there's another closure in the API?

Reply:
Yes, there were two reasons.

- Putting the closure parameters together seems more natural since it's likely they'll be the parameters used most often.
- And, in the wake of the acceptance of [SE-0279](https://github.com/apple/swift-evolution/blob/master/proposals/0279-multiple-trailing-closures.md), there is a new multiple trailing closure syntax. This API is designed to be compatible with that spelling.

Reply:
It looks like this will all parameters:

```swift
session.withAllRequests(completingOn: .main) { (request) in
    // Do something with all requests.
} completion: {
    // Do something when complete.
}
```